src/index.ts

```diff
@@ -386,9 +386,54 @@ export async function install(
     chunkFileNames: 'common/[name]-[hash].js',
   };
   if (Object.keys(depObject).length > 0) {
-    const packageBundle = await rollup.rollup(inputOptions);
-    await packageBundle.write(outputOptions);
+    // Main Snowpack Install 🚀
+    try {
+      const packageBundle = await rollup.rollup(inputOptions);
+      await packageBundle.write(outputOptions);
+    } catch (err) {
+      // Handle Snowpack Install error
+      const invalidFile = err.loc.file as string;
```

Comment:
So some interesting trivia I learned about Rollup: the error handling is ✨ amazing ✨ 

There are actually 2 things we could use here:

- `err.loc` which gives you a detailed breakdown of the exact file & line number the error occurred on
- `err.watchFiles` gives you an array of every single file Rollup has scanned up until the error.

I initially started with `watchFiles` first, and was going to give the ability to show multiple suggestions at once. But I ended up going with `err.loc.file` for 2 reasons:

1. `err.loc.file` gives you the file that erred, so there’s no confusion what the problem was
1. with `err.watchFiles`, you have to write some more logic parsing & massaging that array, getting the file extensions, getting non-JS file extensions, deduping… at the end of it, you are only left with the files Rollup scanned _up until the error_, which means there’s a high probability the only thing that erred was `err.loc.file` to begin with.

To confirm this I tried adding a `.css` and `.svelte` file to a test, but every time Rollup erred, only one of them appeared in `watchFiles` because it threw an error before it kept traversing.

---

All that to say: there’s a lot of smart stuff we can do with this `err` object, and I found `err.loc.file` for this problem to be the simplest. But it’s by no means the only way to solve it.

Reply:
This is great, thank you for sharing your thinking here (makes sense to me). Can you summarize this into a few lines of JS comments and add it to the codebase, for posterity and for the next person who looks at this?